src/client/Microsoft.Identity.Client/ManagedIdentity/AbstractManagedIdentity.cs

```diff
@@ -141,5 +135,52 @@ internal string GetMessageFromErrorResponse(HttpResponse response)
 
             return $"[Managed Identity] Error Code: {managedIdentityErrorResponse.Error} Error Message: {managedIdentityErrorResponse.ErrorDescription}";
         }
+
+        private void HandleException(Exception ex, 
+            ManagedIdentitySource managedIdentitySource = ManagedIdentitySource.None, 
+            string additionalInfo = null)
+        {
+            ManagedIdentitySource source = managedIdentitySource != ManagedIdentitySource.None ? managedIdentitySource : _sourceType;
+
+            if (ex is HttpRequestException httpRequestException)
+            {
+                CreateAndThrowException(MsalError.ManagedIdentityUnreachableNetwork, httpRequestException.Message, httpRequestException, source);
+            }
+            else if (ex is TaskCanceledException)
+            {
+                _requestContext.Logger.Error(TimeoutError);
+            }
+            else if (ex is FormatException formatException)
+            {
+                string errorMessage = additionalInfo ?? formatException.Message;
+                _requestContext.Logger.Error($"[Managed Identity] Format Exception: {errorMessage}");
+                CreateAndThrowException(MsalError.InvalidManagedIdentityEndpoint, errorMessage, formatException, source);
+            }
+            else if (ex is not MsalServiceException or TaskCanceledException)
+            {
+                _requestContext.Logger.Error($"[Managed Identity] Exception: {ex.Message}");
+                CreateAndThrowException(MsalError.ManagedIdentityRequestFailed, ex.Message, ex, source);
+            }
+            else
+            {
+                // If it's already a MsalServiceException, rethrow it
+                throw ex;
```

Comment:
You don't need to rethrow the exception here since you are already doing it at line 70 after the HandleException is called.